File: packages/api/src/services/authService.js
import bcrypt from 'bcryptjs';
import jwt from 'jsonwebtoken';
import { config, usersRepo } from '@sports/shared';
import { httpError } from '../middleware/errorHandler.js';

const BCRYPT_COST = 12;
const TOKEN_TTL = '7d';

export async function signup({ email, password }) {
  const existing = await usersRepo.findUserByEmail(email);
  if (existing) throw httpError(409, 'Email already registered');
  const passwordHash = await bcrypt.hash(password, BCRYPT_COST);
  const user = await usersRepo.createUser({ email, passwordHash });
  return { id: user.id, email: user.email };
}

export async function login({ email, password }) {
  const user = await usersRepo.findUserByEmail(email);
  if (!user) throw httpError(401, 'Invalid credentials');
  const ok = await bcrypt.compare(password, user.password_hash);
  if (!ok) throw httpError(401, 'Invalid credentials');
  return { id: user.id, email: user.email };
}

export function signToken(userId) {
  return jwt.sign({ sub: userId }, config.jwtSecret, { expiresIn: TOKEN_TTL });
}

export function authCookieOptions() {
  return {
    httpOnly: true,
    secure: config.env === 'production',
    sameSite: 'lax',
    maxAge: 7 * 24 * 60 * 60 * 1000,
    path: '/',
  };
}

File: backend/app/services/auth_service.py
from __future__ import annotations

from fastapi import HTTPException, status

from app.repositories import users as users_repo
from app.security import hash_password, verify_password


async def signup(email: str, password: str) -> dict:
    if await users_repo.find_user_by_email(email):
        raise HTTPException(status.HTTP_409_CONFLICT, "Email already registered")
    user = await users_repo.create_user(email, hash_password(password))
    return {"id": user["id"], "email": user["email"]}


async def login(email: str, password: str) -> dict:
    user = await users_repo.find_user_by_email(email)
    if not user or not verify_password(password, user["password_hash"]):
        raise HTTPException(status.HTTP_401_UNAUTHORIZED, "Invalid credentials")
    return {"id": user["id"], "email": user["email"]}
